CRM/Civiconfig/Entity/CustomGroup.php

```diff
@@ -54,87 +53,49 @@ protected function prepareParams(array &$params, array $existing = []) {
-              $membershipType = new CRM_Civiconfig_Entity_MembershipType();
-              $found = $membershipType->getExisting(['name' => $extendsValue]);
-              if (isset($found['id'])) {
-                $params['extends_entity_column_value'][] = $found['id'];
-              }
-              unset ($membershipType);
-            }
-          } else {
-            $membershipType = new CRM_Civiconfig_Entity_MembershipType();
-            $found = $membershipType->getExisting(['name' => $params['extends_entity_column_value']]);
-            if (isset($found['id'])) {
-              $params['extends_entity_column_value'] = $found['id'];
-            }
-          }
-        }
+      case 'Contribution':
+        $entitySubType = 'FinancialType';
         break;
-      case "Relationship":
-        if (!empty($params['extends_entity_column_value'])) {
-          if (is_array($params['extends_entity_column_value'])) {
-            foreach ($params['extends_entity_column_value'] as $extendsValue) {
-              $relationshipType = new CRM_Civiconfig_Entity_RelationshipType();
-              $found = $relationshipType->getExisting(['name_a_b' => $extendsValue]);
-              if (isset($found['id'])) {
-                $params['extends_entity_column_value'][] = $found['id'];
-              }
-              unset ($relationshipType);
-            }
-          } else {
-            $relationshipType = new CRM_Civiconfig_Entity_RelationshipType();
-            $found = $relationshipType->getExisting(['name_a_b' => $params['extends_entity_column_value']]);
-            if (isset($found['id'])) {
-              $params['extends_entity_column_value'] = $found['id'];
-            }
-          }
-        }
+      case 'ParticipantEventType':
+        $entitySubType = 'EventType';
         break;
-      case "ParticipantEventType":
-        if (!empty($params['extends_entity_column_value'])) {
-          if (is_array($params['extends_entity_column_value'])) {
-            foreach ($params['extends_entity_column_value'] as $extendsValue) {
-              $eventType = new CRM_Civiconfig_Entity_EventType();
-              $found = $eventType->getExisting(['name' => $extendsValue]);
-              if (isset($found['value'])) {
-                $params['extends_entity_column_value'][] = $found['value'];
-              }
-              unset ($eventType);
-            }
-          } else {
-            $eventType = new CRM_Civiconfig_Entity_EventType();
-            $found = $eventType->getExisting(['name' => $params['extends_entity_column_value']]);
-            if (isset($found['value'])) {
-              $params['extends_entity_column_value'] = $found['value'];
-            }
-          }
-        }
+      default:
+        $entitySubType = $entityType . "Type";
         break;
     }
+    $entitySubTypeKey = in_array($entityType, [
+      'Activity',
+      'Event'
+    ]) ? 'value' : 'id';
```

Comment:
ParticipantEventType also works based on the value, not the id. 'ParticipantEventType' should also be included in the in_array statement